tests/rptest/tests/full_disk_test.py

```diff
@@ -0,0 +1,118 @@
+# Copyright 2022 Redpanda Data, Inc.
+#
+# Use of this software is governed by the Business Source License
+# included in the file licenses/BSL.md
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0
+import random
+from time import sleep, time
+
+from ducktape.cluster.cluster import ClusterNode
+from ducktape.tests.test import TestContext
+from ducktape.utils.util import wait_until
+from rptest.clients.types import TopicSpec
+from rptest.services.admin import Admin
+from rptest.services.cluster import cluster
+from rptest.tests.redpanda_test import RedpandaTest
+from rptest.utils.node_metrics import NodeMetrics
+from rptest.util import wait_until_result
+
+from kafka import KafkaProducer
+
+N = 3
+# reduce this?
+MAX_MSG = 500
+MAX_MSG_PER_SEC = 10
+FDT_LOG_ALLOW_LIST = [".*cluster - storage space alert: free space.*"]
+
+
+class FullDiskTest(RedpandaTest):
+
+    TOPIC = "fulldisktest-topic"
```

Comment:
the TopicSpec will choose a random topic name for you `topics = [TopicSpec()]`

Reply:
This is outdated, but I'll add another patch. Kafka-python takes producer name as an arg.